src/StatsdClient/Worker/AsynchronousWorker.cs

```diff
@@ -55,29 +58,41 @@ public bool TryEnqueue(T value)
             return _queue.TryEnqueue(value);
         }
 
-        public void Dispose()
+        public void Flush()
         {
-            var remainingWaitCount = maxWaitDurationInDispose.TotalMilliseconds / MinWaitDuration.TotalMilliseconds;
+            var remainingWaitCount = maxWaitDurationInFlush.TotalMilliseconds / MinWaitDuration.TotalMilliseconds;
             while (_queue.QueueCurrentSize > 0 && remainingWaitCount > 0)
             {
                 _waiter.Wait(MinWaitDuration);
                 --remainingWaitCount;
             }
 
-            _terminate = true;
-            try
+            _requestFlush = true;
+            _flushEvent.WaitOne(maxWaitDurationInFlush);
+        }
+
+        public void Dispose()
+        {
+            if (!_terminate)
```

Comment:
NIT: could flip this to reduce nesting:
`if (_terminate) { return } `

Reply:
I tend to use this pattern where there are several `return` in the function.